tensorflow/core/kernels/mkl_aggregate_ops.cc

```diff
@@ -246,15 +259,15 @@ class MklAddNOp : public OpKernel {
       bool input1_in_mkl_format = input1_shape.IsMklTensor();
       bool input2_in_mkl_format = input2_shape.IsMklTensor();
       dnnDelete_F32(Eltwise);
+      if (!input1_in_mkl_format || !input2_in_mkl_format) {
```

Comment:
This condition should be 

if (input1_in_mkl_format || input2_in_mkl_format) {


To be the opposite of line 81.

Reply:
@rmlarsen this part is the opposite of line 173 and 182, only in the case that input 1 and input 2 are not in mkl format, we will need to allocate the array. so we will release the buffer only in the case these two are not in mkl format. Thank you ~ please let me know if you have any suggestions. :)

Reply:
haha, I guess I didn't look at the fully expanded view :-) Thanks for pointing that out.